modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/NativeLibrariesFileDeletion.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.internal;
+
+import static java.lang.String.format;
+
+import static org.apache.commons.io.FileUtils.deleteDirectory;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.slf4j.Logger;
+
+public class NativeLibrariesFileDeletion {
+
+  private final String applicationName;
+  private final File appNativeLibrariesFolder;
+  private static final Logger LOGGER = getLogger(NativeLibrariesFileDeletion.class);
+
+  public NativeLibrariesFileDeletion(String applicationName, File appNativeLibrariesFolder) {
+    this.applicationName = applicationName;
+    this.appNativeLibrariesFolder = appNativeLibrariesFolder;
+  }
+
+  public boolean doAction() {
+    boolean actionPerformed = true;
+
+    try {
+      deleteDirectory(appNativeLibrariesFolder);
+    } catch (IOException e) {
```

Comment:
Either you catch Throwable here or you need to add a catch for the rest of the possible exceptions at the RetryScheduledFileDeletionTask#performAction to trap all the rest of them.
We should unit test this too.